### PR TITLE
fix(ut): fix a flaky test integration_test.write_corrupt_db

### PR DESCRIPTION
**Pull request by acelyc111**
https://github.com/apache/incubator-pegasus/issues/1383

Commit 9303c3aba87c8392705e52baeeee768ad385d5b3 introduced a flaky test, this
patch try to fix it.

This patch also introduce some integration test utils, they would be helpful
for following patches.

### CODE REVIEW

File: src/test/function_test/utils/test_util.h [hunk 1]
```diff
@@ -55,7 +55,14 @@ class test_util : public ::testing::Test
 
     void SetUp() override;
 
-    void run_cmd_from_project_root(const std::string &cmd);
+    static void run_cmd_from_project_root(const std::string &cmd);
+
+    // Get the count of alive replica servers.
+    static int get_replica_server_count();
```
Comment (levy5307) [inline src/test/function_test/utils/test_util.h:61]: get_alive_replica_count

Comment (acelyc111) [reply src/test/function_test/utils/test_util.h]: Renamed to `get_alive_replica_server_count`

**Review by levy5307 — APPROVED** [review]

File: src/test/function_test/utils/test_util.cpp [hunk 2]
```diff
@@ -88,4 +99,38 @@ void test_util::run_cmd_from_project_root(const std::string &cmd)
...
 }
 
+int test_util::get_alive_replica_server_count()
+{
+    std::string json_filename = fmt::format("test_json_file.{}", dsn::rand::next_u32());
+    auto cleanup =
+        dsn::defer([json_filename]() { dsn::utils::filesystem::remove_path(json_filename); });
+    run_cmd_from_project_root(fmt::format("echo 'nodes -djo {}' | ./run.sh shell", json_filename));
+    std::ifstream f(json_filename);
+    json data = json::parse(f);
```
Comment (empiredan) [inline src/test/function_test/utils/test_util.cpp:109]:
```suggestion
    const auto data = json::parse(f);
```

File: src/test/function_test/utils/test_util.cpp [hunk 3]
```diff
@@ -88,4 +99,38 @@ void test_util::run_cmd_from_project_root(const std::string &cmd)
...
+
+int test_util::get_leader_count(const std::string &table_name, int replica_server_index)
+{
+    std::string json_filename = fmt::format("test_json_file.{}", dsn::rand::next_u32());
+    auto cleanup =
+        dsn::defer([json_filename]() { dsn::utils::filesystem::remove_path(json_filename); });
+    run_cmd_from_project_root(
+        fmt::format("echo 'app {} -djo {}' | ./run.sh shell", table_name, json_filename));
+    std::ifstream f(json_filename);
+    json data = json::parse(f);
```
Comment (empiredan) [inline src/test/function_test/utils/test_util.cpp:125]:
```suggestion
    const auto data = json::parse(f);
```

**Review by empiredan — APPROVED** [review]

**Review by levy5307 — APPROVED** [review]

**Pull request merged**